src/OrchardCore.Modules/OrchardCore.Users/Deployment/LoginSettingsDeploymentSource.cs

```diff
@@ -0,0 +1,38 @@
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using OrchardCore.Deployment;
+using OrchardCore.Entities;
+using OrchardCore.Settings;
+using OrchardCore.Users.Models;
+using OrchardCore.Users.Services;
+
+namespace OrchardCore.Users.Deployment
+{
+    public class LoginSettingsDeploymentSource : IDeploymentSource
+    {
+        private readonly ISiteService _siteService;
+
+        public LoginSettingsDeploymentSource(ISiteService siteService)
+        {
+            _siteService = siteService;
+        }
+
+        public async Task ProcessDeploymentStepAsync(DeploymentStep step, DeploymentPlanResult result)
+        {
+            var loginSettingsStep = step as LoginSettingsDeploymentStep;
+
+            if (loginSettingsStep == null)
+            {
+                return;
+            }
+
+            var loginSettings = (await _siteService.GetSiteSettingsAsync()).As<LoginSettings>();
+
```

Comment:
Yes, like this. Add a null check, and don't add the settings if they are null. would be pointless